src/front/wgsl/mod.rs

```diff
@@ -190,6 +340,49 @@ impl<'a> Error<'a> {
     }
 }
 
+fn type_to_name(tp: &crate::TypeInner) -> String {
```

Comment:
We can also do `impl crate::TypeInner { fn to_name(&self) {...} }` or something like this.

Reply:
But `TypeInner` is not a wgsl specific type, but this name is the name used in wgsl. It didn't feel quite right to put wgsl-specific formatting code outside the wgsl directory.

Reply:
It's totally cool because the method would be private, only visible to WGSL. We do this in other frontends/backends.

Reply:
Ok. I added an `to_wgsl` function which is quite nice.
Eventually we might want to re-use some printing code from wgsl-out. But I'm not sure how easy that is to do right now.